rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -133,19 +132,6 @@ private void initialize(Configuration conf, BookKeeperFactory bookKeeperFactory)
     this.blockSize = CacheConfig.getBlockSize(conf);
     this.localPath = CacheConfig.getLocalPath(remotePath, conf);
     this.diskReadBufferSize = CacheConfig.getDiskReadBufferSizeDefault(conf);
-    File file = new File(localPath);
-    if (!file.exists()) {
```

Comment:
There were multiple commits in git blame in the removed code so there is some history around this piece. Can there be side effects of removing this? e.g some other class might be expecting file to be already there because of this code or expects right read/write perms set by this code

Reply:
This code is creating a file if the file doesn't exist. This code is running at the client side (not in bookkeeper) and at that time the client has no idea whether this node should have the file or not. So thats why this piece of code was creating the file unnecessarily. If Rubix decides that it needs cache a file from remote file system, it will go through the RemoteReadRequestChain where we are checking whether the corresponding local file exists or not. If not we are creating the then. So this piece of code is redundant.

This is from RemoteReadRequestChain
```
public Integer call()
      throws IOException
  {
    Thread.currentThread().setName(threadName);
    checkState(isLocked, "Trying to execute Chain without locking");

    if (readRequests.size() == 0) {
      return 0;
    }

    // Issue-53 : Open file with the right permissions
    File file = new File(localFile);
    if (!file.exists()) {
      file.createNewFile();
      file.setWritable(true, false);
      file.setReadable(true, false);
    }
```